tools/fdr2csv/src/main.rs

```diff
@@ -0,0 +1,174 @@
+    let res = bytemuck::from_bytes::<T>(buf.as_slice());
+
+    Ok(*res)
+}
+
+fn main() -> Result<(), std::io::Error> {
+    // Parse CLI arguments
+    let args = Args::parse();
+
+    // Open the input file
+    let in_file = File::open(args.input.trim())
+        .map_err(|e| std::io::Error::new(e.kind(), "Failed to open input file!"))?;
+
+    // Create Gzip Reader
+    let mut reader: Box<dyn Read> = if args.no_compression {
+        Box::new(BufReader::new(in_file))
+    } else {
+        Box::new(GzDecoder::new(BufReader::new(in_file)))
+    };
+
+    // Read file version
+    let file_format_version = read_bytes::<u64>(&mut reader)?;
+    let aircraft_type = if file_format_version > a380::INTERFACE_MIN_VERSION {
+        AircraftType::A380
+    } else {
+        AircraftType::A320
+    };
+
+    let aircraft_interface_version = match aircraft_type {
+        AircraftType::A320 => a320::INTERFACE_VERSION,
+        AircraftType::A380 => a380::INTERFACE_VERSION,
+    };
+
+    // Print or check file version
+    if args.get_input_file_version {
+        println!(
+            "Aircraft Type is {:?}, Interface version is {}",
+            aircraft_type, file_format_version
+        );
+        return Ok(());
+    } else if aircraft_interface_version != file_format_version {
+        return Err(std::io::Error::new(
+            ErrorKind::InvalidInput,
+            format!(
+                "Mismatch between converter and file version (expected {aircraft_interface_version}, got {file_format_version})",
+            ),
+        ));
+    }
+
+    // Print info on conversion start
+    println!(
+        "Converting from '{}' to '{}' for aircraft type '{:?}' with interface version '{}' and delimiter '{}'",
+        args.input, args.output, aircraft_type, file_format_version, args.delimiter
+    );
+
+    // Open or create output file in truncate mode
+    let out_file = OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .create(true)
+        .open(args.output.trim())
+        .map_err(|e| std::io::Error::new(e.kind(), "Failed to open output file!"))?;
+
+    let mut buf_writer = BufWriter::new(out_file);
+
+    let mut counter = 0;
+
+    // Generate and write the header
+    let header = match aircraft_type {
+        AircraftType::A320 => {
+            csv_header_serializer::to_string(&a320::FdrData::default(), args.delimiter)
+        }
+        AircraftType::A380 => {
+            csv_header_serializer::to_string(&a380::FdrData::default(), args.delimiter)
+        }
+    }
+    .map_err(|_| std::io::Error::new(ErrorKind::Other, "Failed to generate header."))?;
+
+    buf_writer.write(header.as_bytes())?;
+
+    // Create the CSV writer, and serialize the file.
+    let mut writer = WriterBuilder::new()
+        .delimiter(args.delimiter as u8)
+        .has_headers(false)
+        .from_writer(buf_writer);
+
+    match aircraft_type {
+        AircraftType::A320 => {
```

Comment:
If any rustaceans have feedback on how to do this better, I'd love to hear it

Reply:
I think you might be able to define a `read_record_function` function pointer 
```
let read_record_function = match aircraft_type { AircraftType::A320 =>  {a320::read_record}, AircraftType::A380 => { a380::read_record } }
```
Then use that in your common workflow. Another approach could be to define a `read_record` function in your AircraftType that returns the right value depending on the variant.

Reply:
I tried defining the function as a variable, but it doesn't work as the return types of the functions are different

Reply:
And defining the return type as Box<dyn Serialize> also doesn't work:
```
    |
54  | pub fn read_record(reader: &mut impl Read) -> Result<Box<dyn Serialize>, Error> {
    |                                                          ^^^^^^^^^^^^^ `a320::_::_serde::Serialize` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
```